### ISSUE TITLE
[PoSV] transaction index out of sync

### ISSUE THREAD

**Issue by laudney** [reddcoin-project/reddcoin-3.10#10]
sometimes the transaction index (as enabled txindex=1 by default in PoSV wallet but disabled txindex=0 in current wallets) is out of sync. when a coinstake transaction in a new PoSV block is conflicted or rejected, the coinstake transaction is not reverted from the tx index, which makes QT wallet think they are lost: inputs are consumed by output is not confirmed.

one can fix this issue by restart your wallet with "-rescan" to force the tx index to be rebuilt from blockchain and one will recover all your coins.

in short, no coin is lost but GUI lies.
